### DISCUSSION TITLE
Testing sequential Prisma Client operations inside a transaction using Jest

### DISCUSSION THREAD

**Discussion by arielclement** [prisma/prisma#22865] [Q&A] [answered]
### Question

Hello everyone, I'm trying to mock a sequential prisma operation inside a transaction for my application. Can anyone please show me how I can do it?

```js
const [posts, totalPosts] = await prisma.$transaction([
  prisma.post.findMany({ where: { title: { contains: 'prisma' } } }),
  prisma.post.count(),
])
```

I am currently using prisma transaction for pagination purposes and my code closely resembles the above. Your feedback would be appreciated.

Many thanks in advance!

**Comment by ludralph** [comment, marked as answer]
Hi @arielclement  👋 

Thank you for raising this question.

You can mock a transaction like this in your test file.
```js
test('mocks sequential Prisma operation inside a transaction', async () => {
  // Mocked data
  const mockedPostsData = [{ id: 1, title: 'Post with prisma' }];
  const mockedCount = 1;

  prismaMock.$transaction.mockResolvedValueOnce([
    mockedPostsData, // Mocking the first operation
    mockedCount // Mocking the second operation
  ]);

  // Invoke the function
  const result = await getUsersAndCount();


  // Assertions to validate the result
  expect(result.posts).toEqual(mockedPostsData);
  expect(result.totalPosts).toEqual(mockedCount);
});
```

prismaMock is defined like shown below:
```js
const { mockDeep, mockReset } = require("jest-mock-extended")

const { prisma }  = require("./client")

jest.mock("./client", () => ({
  prisma: mockDeep()
}))

beforeEach(() => {
  mockReset(prismaMock)
})

const prismaMock = prisma

module.exports = {
    prismaMock
}
```

The client file is shown below:
```js
const { PrismaClient } = require('@prisma/client')

const prisma = new PrismaClient()

module.exports = {
    prisma
}
```

If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

**Reply by arielclement** [reply]
Thanks @ludralph!

**Reply by ludralph** [reply]
You're welcome 😄
